### DISCUSSION TITLE
umami v2 converted numeric IDs to VARCHAR(36) on MySQL (bad idea?)

### DISCUSSION THREAD

**Discussion by juangacovas** [umami-software/umami#2013] [General]
Loving version 2 after migrating, but noticed that some sites with +2M hits started to give errors when loading stats.

Queries would take some seconds on v1 but longer and longer under v2, hitting errors (unable to load). Even trying to delete or reset website data hangs. My setup is MariaDB 10.11.2

Then noticed the table structured changed numeric INTs for VARCHAR(36) on all ID columns! Properly indexed but... perhaps this is fine in PostgreSQL with special columns that are internally numeric, but... really? IDs should be numeric for performance

Any thoughts?

**Comment by MarcHagen** [comment]
The new data structure uses UUIDs (like postgres) as `id`. So varchar is not unexpected in this scenario.

**Reply by juangacovas** [reply]
Hello. I'm using MariaDB and saw slow queries. I recently converted all umami "*_id" columns in all tables to "UUID" data type (and NOT varchar 36) and not only saved GB of space, also query speed was good too. So it's possible to get nice speed with mysql and NOT having to switch to [PostgreSQL](https://www.postgresql.org/)...

UUID data type is supported from MariaDB version 10.7.0. See https://mariadb.com/kb/en/uuid-data-type/

**Reply by juangacovas** [reply]
See also [this](https://github.com/umami-software/umami/discussions/2675)
